# <h1><center><span style="color:yellow">**Bismillahirrahmanirrahim**</span></center></h1>
# <h1><center><span style="color:white">Allahummayasir walatu'asir</span></center></h1>
# <center><span style='color:white'>**TUGAS 6**</span></center>


<h2>Soal 1: Membuat Objek DataFrames</h2>

- Buatlah objek dataframes dari objek dictionary, Bebas!

In [1]:
# Membuat dataframes dari dictionary
import pandas as pd
data_frame_ku = pd.DataFrame(
                            {'Nama siswa':['Ahmad','Sulaiman','Muhammad','Saiful','udin','joko'],
                            'Nilai siswa':[90,87,89,92,87,76]}, index=[1,2,3,4,5,6]
                            )
display(data_frame_ku)

dict_nilai_siswa = {'Nama siswa':['Ahmad','Sulaiman','Muhammad','Saiful','udin','joko'],
                    'Nilai siswa':[90,87,89,92,87,76]}
data_frame_ku_2 = pd.DataFrame(dict_nilai_siswa, index=[1,2,3,4,5,6])
display(data_frame_ku_2)


Nama siswa  Nilai siswa
1      Ahmad           90
2   Sulaiman           87
3   Muhammad           89
4     Saiful           92
5       udin           87
6       joko           76

Nama siswa  Nilai siswa
1      Ahmad           90
2   Sulaiman           87
3   Muhammad           89
4     Saiful           92
5       udin           87
6       joko           76

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian print.

In [3]:
data_diamonds = pd.read_csv('diamonds-1.csv')
display(data_diamonds)
# Membaca data dan memasukannya kedalam suatu variabel


Unnamed: 0  carat        cut color clarity  depth  table  price     x  \
0               1   0.23      Ideal     E     SI2   61.5   55.0    326  3.95   
1               2   0.21    Premium     E     SI1   59.8   61.0    326  3.89   
2               3   0.23       Good     E     VS1   56.9   65.0    327  4.05   
3               4   0.29    Premium     I     VS2   62.4   58.0    334  4.20   
4               5   0.31       Good     J     SI2   63.3   58.0    335  4.34   
...           ...    ...        ...   ...     ...    ...    ...    ...   ...   
53935       53936   0.72      Ideal     D     SI1   60.8   57.0   2757  5.75   
53936       53937   0.72       Good     D     SI1   63.1   55.0   2757  5.69   
53937       53938   0.70  Very Good     D     SI1   62.8   60.0   2757  5.66   
53938       53939   0.86    Premium     H     SI2   61.0   58.0   2757  6.15   
53939       53940   0.75      Ideal     D     SI2   62.2   55.0   2757  5.83   

          y     z  
0      3.98  2.43  
1      3.84  2.31  
2      4.07  2.31  
3      4.23  2.63  
4      4.35  2.75  
...     ...   ...  
53935  5.76  3.50  
53936  5.75  3.61  
53937  5.68  3.56  
53938  6.12  3.74  
53939  5.87  3.64  

[53940 rows x 11 columns]

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: DataFrame vs Series</h2>

- Apa perbedaan Series dengan list dan dictionary
- Apa perbedaan dataframe dan series
- Buatlah suatu data series

Jawab: 
- Series adalah seperti gabungan dari list dan dictionary dimana dalam list kita bisa mengakses/indexing pada object series dan memiliki index yang berguna seperti keys pada dictionary. Series dapat menyimpan berbagai tipe data seperti int, float, dll sama seperti list namun memliki index.
- DataFrame adalah suatu objek 2 dimensi yang memiliki 2 index yaitu baris dan column, sedangkan series adalah objek 1 dimensi yang hanya memiliki 1 index.

In [44]:
# code here
dict1 = {'nama':['acan','joko'],'kelas':['7a','7b']}
data_series_dict = pd.Series(dict1)
display(data_series_dict)

list_1 = ['acan','joko','udin','tomo']
list_2 = ['7a','7b','7c','7d']
data_series_list = pd.Series(list_1,list_2)
display(data_series_list)



data_series = pd.Series(range(11), index=[1,2,3,4,5,6,7,8,9,10,11])
display(data_series)

nama     [acan, joko]
kelas        [7a, 7b]
dtype: object

7a    acan
7b    joko
7c    udin
7d    tomo
dtype: object

1      0
2      1
3      2
4      3
5      4
6      5
7      6
8      7
9      8
10     9
11    10
dtype: int64

<h2>Soal 4: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- baris pertama dari column price
- baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [18]:
# code here
print(data_diamonds['price'].head(1))
print(data_diamonds['color'].tail(1))

0    326
Name: price, dtype: int64
53939    D
Name: color, dtype: object


<h2>Soal 5: Deskriptif Statistik</h2>

Jelaskan apa itu deskriptis statistik!

Jawab: 
- Statistika deskriptif adalah suatu metode dalam ilmu statistika untuk melihat gambaran umum dari suatu data. Metode ini digunakan sebelum kita melakukan metode statistika inferensia dan biasanya digunakan dalam tahap eksplorasi data. Disajikan bisa dalam statistika 5 serangkai yaitu terdiri dari nilai (Min, Q31, Nilai tengah, Q3, Max). Lalu dilihat juga ukuran pemusatan suatu data yang dapat dilihat dari nilai Mean, Median, Modus dan penyebaran suatu data yang bisa dilihat dari nilai standar deviasi dan ragam. Pada metode ini dapat dianalisa juga apakah data penyebar normal atau tidak, apakah ada skewness/menjorok, dan apakah ada nilai pencilan. Dapat disajikan berupa tabel ataupun dibuat grafik-grafik seperti histogram, boxplot, dll. 

<h2>Soal 6: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [17]:
# code here
display(data_diamonds['price'].mean())
display(data_diamonds['depth'].std())
display(data_diamonds['carat'].max())

3932.799721913237

1.4326213188337733

5.01

Jawab: 
- Nilai rata-rata dari column price adalah 3932.799
- Nilai standar deviasi dari column depth adalah 1.432
- Nilai maksimum dari column carat adalah 5.01